numpy/lib/histograms.py

```diff
@@ -318,9 +318,15 @@ def _get_bin_edges(a, bins, range, weights):
         raise ValueError('`bins` must be 1d, when an array')
 
     if n_equal_bins is not None:
+        # gh-10322 forces us to request this now to avoid inconsistency
```

Comment:
Here and in the other comment below, ideally explain in a sentence why we are doing this, and then add "see gh-10322" - future developers will thank you!